apps/frontend/core-dapp/src/features/portfolio/PortfolioStore.ts

```diff
@@ -147,32 +65,29 @@ export class PortfolioStore {
     }
   }
 
-  get isLoadingPositions(): boolean {
-    if (!this.root.web3Store.connected) return false
-    // find any position that is still loading data
-    const loadingIndex = this.positions.findIndex(({ data }) => data === undefined)
-    // loading is true if loadingIndex is a valid index, (e.g. 0, 1, 2, ...)
-    return loadingIndex > -1
-  }
-
-  get tradingPositions(): Position[] {
+  // all possible positions including those that user has 0 balance
+  get allPositions(): PositionEntity[] {
     const { marketStore } = this.root
     const { markets } = marketStore
 
-    const positions: Position[] = []
+    const positions: PositionEntity[] = []
     Object.values(markets).forEach((market) => {
-      const longPosition = this.hasPosition(market, 'long')
-      const shortPosition = this.hasPosition(market, 'short')
-
-      if (longPosition) positions.push(longPosition)
-      if (shortPosition) positions.push(shortPosition)
+      positions.push(new PositionEntity(this.root, market, 'long'))
+      positions.push(new PositionEntity(this.root, market, 'short'))
     })
 
     return positions
   }
 
-  get positions(): Position[] {
-    return [...this.tradingPositions]
+  // only positions where user has more than 0 balance
+  get positions(): PositionEntity[] | undefined {
```

Comment:
rename to userPositions
```suggestion
  get userPositions(): PositionEntity[] | undefined {
```